neuron_morphology/transforms/geometry.py

```diff
@@ -13,6 +13,49 @@ def get_ccw_vertices_from_two_lines(line1: List[Tuple], line2: List[Tuple]):
     """
     return get_ccw_vertices(get_vertices_from_two_lines(line1, line2))
 
+def prune_two_lines(line1: List[Tuple], line2: List[Tuple]):
+    """
+        check the boundary to avoid intersections with side lines
+
+        Parameters
+            ----------
+            line1, line2: List of coordinates describing two lines
+
+            Returns
+            -------
+            line1, line2: boundary pruned if needed
+    """
+    # validate the pia/wm does not cross the side lines
+    prune = True
+    while prune:
+        prune = False
+
+        # update lines
+        line1_str = geo.LineString(line1)
+        line2_str = geo.LineString(line2)
+
+        side1 = geo.LineString([line1[0], line2[-1]])
+        side2 = geo.LineString([line1[-1], line2[0]])
+
+        # prune the edge points if there are intersections of side and pia/wm
+        if side1.crosses(line1_str):
+            line1.pop(0)
+            prune = True
+
+        if side1.crosses(line2_str):
+            line2.pop(-1)
+            prune = True
+
+        if side2.crosses(line1_str):
+            line1.pop(-1)
+            prune = True
+        
+        if side2.crosses(line2_str):
+            line2.pop(0)
+            prune = True
+
```

Comment:
You probably want to issue a warning if boundary was modified

Reply:
added